Update Guide “pacing-progression” (#131)
komatic5/gdcs2: content/docs/guides/gameplay-2/pacing-progression.md

* Important aspects to induce flow are; Having Clear Goals and Rewards; Rapid Feedback; and Adequate Challenges.
* Micro flow is the short-term version of flow, occurring on a second to second basis in gameplay. Important factors to creating good micro flow are Rhythm, Encouragement, and Positive Feedback.
{{< /callout >}}

- - -

# 1: The Flow State

When we describe Pacing, another term related to it is the **Flow State**. The Flow State is a **mental state where you put intense concentration into one task and lose focus on everything else**. You may have heard it referred to as “the zone” before.

In terms of gameplay, it’s very effective in making the player enjoy your level. Research suggests that when you’re in the zone, you’re more persistent at what you’re doing and have higher dopamine levels. This persistence and willingness to engage with the level are exactly what we want our pacing to accomplish.

Pacing is critical to inducing flow and getting and maintaining the player’s attention. There are three factors that are crucial for flow that you need to know:

> * **Clear Goals & Rewards:** The player should know what to do to not die, and why they died when they do die. This doesn't apply when learning a level, but if a player doesn't know why they died even after learning the level the outcome of their actions feels random. Making clear what mistake a player made is often critical for maintaining flow.

Reviewer: ```suggestion
* **Clear Goals & Rewards:** When playing a level, the player *should have a good understanding of what to do to complete it*. The player should know what to do to not die, and why they died when they *do* die. If a player doesn't know why they died, even after learning the level, the outcome of their actions will feel random and the level will feel unfair. Making clear what mistake a player made is often critical for maintaining flow. Additionally, the player should get rewarded for making progress towards the goal of the level.
```

> * **Rapid Feedback:** Ideally, an incorrect input should kill you quickly so that you can more easily associate that death with the incorrect input

Reviewer: ```suggestion
* **Rapid Feedback:** You should know immediately if your actions will have a good effect or a bad one. Ideally, *an incorrect input should kill you quickly* so that you can more easily associate that death with the incorrect input. This helps a lot with learning what to do and what to avoid.
```

> * **Adequate Challenge:** Typically this is more the responsibility of the player choosing a level for their skill level, you can't make a level for everyone. That being said, maintaining a reasonably consistent difficulty within the level is generally ideal.

Reviewer: ```suggestion
* **Adequate Challenge:** There should be a good balance between your skill and the level’s difficulty. This helps prevent it from being too boring or too stressful. Typically this is more the responsibility of the player choosing a level for their skill level, as you can't make a level for everyone, but maintaining a reasonably consistent difficulty within the level is generally ideal.
```

# 2: Micro Flow

**Micro flow** is a form of short-term flow that occurs from one second to the next as you play. It’s the main consideration when you’re *building* your gameplay.

As you may expect, the prior factors – clear goals, rapid feedback, and adequate challenge – are important to develop good pacing at this stage. However, there are some other factors that are important for micro flow.

> * **Rhythm:** Try to design the level’s clicks to fit a pattern. Repetition is a powerful tool that makes the brain more responsive to flow. It’s also part of why sync is so effective, but that’s a discussion for the Sync & Rhythm guide.

Reviewer: ```suggestion
* **Rhythm:** Try to design the level’s clicks to fit a pattern. Repetition is a powerful tool that makes the brain more responsive to flow. It’s also part of why sync is so effective, but that’s a discussion for the [Note Representation](/docs/guides/gameplay-2/note-representation/) guide.
```

> * **Encouragement**: Your level should give the feeling whatever is done makes a sense of progression. This means that as the level creator, you can sometimes cheat the player a little bit and give them more room for error. This will be discussed more in the Fairness guide.

Reviewer: ```suggestion
* **Encouragement**: Your level should give the feeling whatever is done makes a sense of progression. This means that as the level creator, you can sometimes cheat the player a little bit and give them more room for error. This will be discussed more in the [Fairness](/docs/guides/gameplay-2/fairness/) guide.
```
